### PR TITLE
feat: Added method sort_columns in Table and changed equals to not care about order

### LINKED ISSUE #305: The equals method in Table (and Row) should not care about the order of the columns
### Is your feature request related to a problem? Please describe

If the order of columns is changed but the content of the table stays the same the equals method from Table returns `false`.

### Desired solution

The order of the columns should not matter. It would be good to add a method `sort_columns(lambda: Callable[[Column, Column], int]): Table` in Table to reorder the columns and use this function in the equals method. Default is alphabetical column names.

### Possible alternatives (optional)

You could also just compare the Tables without considering the order of the columns

### PR DESCRIPTION
**Pull request by dorianheidorn**
Closes #305 .

### Summary of Changes

Added method sort_columns in Table and changed equals to not care about order of columns. The sort_columns method by default sorts after alphabetically order which can be changed by porividing a lambda-function as parameter.

### CODE REVIEW

**Review by WinPlay02 — CHANGES_REQUESTED** [review]
Blocked

**Review by WinPlay02 — APPROVED** [review]

File: Runtime/safe-ds/safe_ds/data/_table.py [hunk 1]
```diff
@@ -612,12 +618,43 @@ def list_columns_with_non_numerical_values(self) -> list[Column]:
...
+        self,
+        query: Callable[[Column, Column], int] = lambda col1, col2: (
+            col1.name > col2.name
+        )
+        - (col1.name < col2.name),
+    ) -> Table:
+        """
+        Sort a Table with the given lambda function.
+        If no function is given the columns will be sorted alphabetically.
+        This function uses bubble sort.
```
Comment (WinPlay02) [inline Runtime/safe-ds/safe_ds/data/_table.py:631]:
```suggestion
        This function uses the default python sort algorithm.
```

File: Runtime/safe-ds/safe_ds/data/_table.py [hunk 2]
```diff
@@ -612,12 +618,43 @@ def list_columns_with_non_numerical_values(self) -> list[Column]:
...
 
+    def sort_columns(
+        self,
+        query: Callable[[Column, Column], int] = lambda col1, col2: (
+            col1.name > col2.name
+        )
+        - (col1.name < col2.name),
+    ) -> Table:
+        """
+        Sort a Table with the given lambda function.
```
Comment (WinPlay02) [inline Runtime/safe-ds/safe_ds/data/_table.py:629]:
```suggestion
        Sort a table with the given lambda function.
```

**Review by WinPlay02 — COMMENTED** [review]
Please update docs

File: Runtime/safe-ds/safe_ds/data/_table.py [hunk 3]
```diff
@@ -612,12 +618,43 @@ def list_columns_with_non_numerical_values(self) -> list[Column]:
...
+        If no function is given the columns will be sorted alphabetically.
+        This function uses bubble sort.
+        The query should return:
+            TRUE if both columns should be switched
+            FALSE if the columns should not be switched
```
Comment (WinPlay02) [inline Runtime/safe-ds/safe_ds/data/_table.py:634]:
```suggestion
        The query should return:
            0, if both columns are equal
            < 0, if the first column should be ordered after the second column
            > 0, if the first column should be ordered before the second column
```

**Review by Gerhardsa0 — DISMISSED** [review]

File: Runtime/safe-ds/safe_ds/data/_table.py [hunk 4]
```diff
@@ -267,7 +270,8 @@ def get_column(self, column_name: str) -> Column:
         raise UnknownColumnNameError([column_name])
 
     def drop_columns(self, column_names: list[str]) -> Table:
-        """Returns a Table without the given columns
+        """
+        Returns a Table without the given columns
```
Comment (WinPlay02) [inline Runtime/safe-ds/safe_ds/data/_table.py:274]:
```suggestion
        Returns a table without the given columns
```

File: Runtime/safe-ds/safe_ds/data/_table.py [hunk 5]
```diff
@@ -300,7 +304,8 @@ def drop_columns(self, column_names: list[str]) -> Table:
         return Table(transformed_data)
 
     def keep_columns(self, column_names: list[str]) -> Table:
-        """Returns a Table with exactly the given columns
+        """
+        Returns a Table with exactly the given columns
```
Comment (WinPlay02) [inline Runtime/safe-ds/safe_ds/data/_table.py:308]:
```suggestion
        Returns a table with exactly the given columns
```

File: Runtime/safe-ds/safe_ds/data/_table.py [hunk 6]
```diff
@@ -346,7 +351,8 @@ def to_rows(self) -> list[Row]:
         ]
 
     def filter_rows(self, query: Callable[[Row], bool]) -> Table:
-        """Returns a Table with rows filtered by applied lambda function
+        """
+        Returns a Table with rows filtered by applied lambda function
```
Comment (WinPlay02) [inline Runtime/safe-ds/safe_ds/data/_table.py:355]:
```suggestion
        Returns a table with rows filtered by applied lambda function
```

File: Runtime/safe-ds/safe_ds/data/_table.py [hunk 7]
```diff
@@ -239,7 +241,8 @@ def rename_column(self, old_name: str, new_name: str) -> Table:
         return Table(new_df.rename(columns={old_name: new_name}))
 
     def get_column(self, column_name: str) -> Column:
-        """Returns a new instance of Column with the data of the described column of the Table.
+        """
+        Returns a new instance of column with the data of the described column of the Table.
```
Comment (WinPlay02) [inline Runtime/safe-ds/safe_ds/data/_table.py:245]:
```suggestion
        Returns a new instance of column with the data of the described column of the table.
```

**Review by WinPlay02 — APPROVED** [review]

**Review by WinPlay02 — APPROVED** [review]

**Pull request merged**